### PR TITLE
deploy commander using helm

### PR DESCRIPTION
**Pull request by willcl-ark**
- deploy commanders using helm chart
- bring them down using helm chart
- speed up bringing a whole warnet down using a limited threadpool, and uninstall everythign from helm properly

### CODE REVIEW

**Comment by willcl-ark** [comment]
God damn CI... those all work for me locally:

<details>
<summary>Details</summary>

```
will@ubuntu in ~/src/warnet on  helm-commander [$?] is 📦 v1.0.0 : 🐍 (warnet)
₿ ./test/dag_connection_test.py
2024-09-08 22:13:24 | INFO    | test     | Logging started
2024-09-08 22:13:24 | INFO    | test     | Warnet test base initialized
2024-09-08 22:13:24 | INFO    | test     | Setting up network
2024-09-08 22:13:24 | DEBUG   | test     | Executing warnet command: deploy /home/will/src/warnet/test/data/ten_semi_unconnected
2024-09-08 22:13:26 | INFO    | test     | Deploying node: tank-0000
Release "tank-0000" does not exist. Installing it now.
NAME: tank-0000
LAST DEPLOYED: Sun Sep  8 22:13:24 2024
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
Thank you for installing tank-0000.
Deploying node: tank-0001
Release "tank-0001" does not exist. Installing it now.
NAME: tank-0001
LAST DEPLOYED: Sun Sep  8 22:13:24 2024
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
Thank you for installing tank-0001.
Deploying node: tank-0002
Release "tank-0002" does not exist. Installing it now.
NAME: tank-0002
LAST DEPLOYED: Sun Sep  8 22:13:24 2024
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
Thank you for installing tank-0002.
Deploying node: tank-0003
Release "tank-0003" does not exist. Installing it now.
NAME: tank-0003
LAST DEPLOYED: Sun Sep  8 22:13:25 2024
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
Thank you for installing tank-0003.
Deploying node: tank-0004
Release "tank-0004" does not exist. Installing it now.
NAME: tank-0004
LAST DEPLOYED: Sun Sep  8 22:13:25 2024
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
Thank you for installing tank-0004.
Deploying node: tank-0005
Release "tank-0005" does not exist. Installing it now.
NAME: tank-0005
LAST DEPLOYED: Sun Sep  8 22:13:25 2024
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
Thank you for installing tank-0005.
Deploying node: tank-0006
Release "tank-0006" does not exist. Installing it now.
NAME: tank-0006
LAST DEPLOYED: Sun Sep  8 22:13:25 2024
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
Thank you for installing tank-0006.
Deploying node: tank-0007
Release "tank-0007" does not exist. Installing it now.
NAME: tank-0007
LAST DEPLOYED: Sun Sep  8 22:13:26 2024
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
Thank you for installing tank-0007.
Deploying node: tank-0008
Release "tank-0008" does not exist. Installing it now.
NAME: tank-0008
LAST DEPLOYED: Sun Sep  8 22:13:26 2024
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
Thank you for installing tank-0008.
Deploying node: tank-0009
Release "tank-0009" does not exist. Installing it now.
NAME: tank-0009
LAST DEPLOYED: Sun Sep  8 22:13:26 2024
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
Thank you for installing tank-0009.
2024-09-08 22:13:26 | DEBUG   | test     | Waiting for predicate with timeout 1200s and interval 5s
2024-09-08 22:13:26 | INFO    | test     | Waiting for all tanks to reach 'running': {'total': 10, 'running': 5, 'pending': 5}
2024-09-08 22:13:31 | INFO    | test     | Waiting for all tanks to reach 'running': {'total': 10, 'running': 10}
2024-09-08 22:13:31 | DEBUG   | test     | Waiting for predicate with timeout 1200s and interval 5s
Tank tank-0000 peers expected: 0, actual: 0
Tank tank-0001 peers expected: 0, actual: 0
Tank tank-0002 peers expected: 0, actual: 0
Tank tank-0003 peers expected: 0, actual: 0
Tank tank-0004 peers expected: 0, actual: 0
Tank tank-0005 peers expected: 0, actual: 0
Tank tank-0006 peers expected: 0, actual: 0
Tank tank-0007 peers expected: 0, actual: 0
Tank tank-0008 peers expected: 1, actual: 0
Network not connected
Tank tank-0000 peers expected: 0, actual: 0
Tank tank-0001 peers expected: 0, actual: 0
Tank tank-0002 peers expected: 0, actual: 0
Tank tank-0003 peers expected: 0, actual: 0
Tank tank-0004 peers expected: 0, actual: 0
Tank tank-0005 peers expected: 0, actual: 0
Tank tank-0006 peers expected: 0, actual: 0
Tank tank-0007 peers expected: 0, actual: 0
Tank tank-0008 peers expected: 1, actual: 0
Network not connected
Tank tank-0000 peers expected: 0, actual: 0
Tank tank-0001 peers expected: 0, actual: 0
Tank tank-0002 peers expected: 0, actual: 0
Tank tank-0003 peers expected: 0, actual: 0
Tank tank-0004 peers expected: 0, actual: 0
Tank tank-0005 peers expected: 0, actual: 0
Tank tank-0006 peers expected: 0, actual: 0
Tank tank-0007 peers expected: 0, actual: 0
Tank tank-0008 peers expected: 1, actual: 1
Tank tank-0009 peers expected: 0, actual: 0
Network connected
2024-09-08 22:13:44 | INFO    | test     | Running connect_dag scenario
2024-09-08 22:13:44 | DEBUG   | test     | Executing warnet command: run test/data/scenario_connect_dag.py
2024-09-08 22:13:45 | DEBUG   | test     | Waiting for predicate with timeout 300s and interval 5s
2024-09-08 22:13:45 | DEBUG   | test     | Scenario commander-scenarioconnectdag-1725830025 exited with code None
2024-09-08 22:13:50 | DEBUG   | test     | Scenario commander-scenarioconnectdag-1725830025 exited with code 0
2024-09-08 22:13:50 | INFO    | test     | Stopping network
2024-09-08 22:13:50 | DEBUG   | test     | Executing warnet command: down
2024-09-08 22:13:51 | DEBUG   | test     | Waiting for predicate with timeout 60s and interval 1s
2024-09-08 22:13:51 | INFO    | test     | Stopping network
2024-09-08 22:13:51 | DEBUG   | test     | Executing warnet command: down
2024-09-08 22:13:51 | DEBUG   | test     | Waiting for predicate with timeout 60s and interval 1s

will@ubuntu in ~/src/warnet on  helm-commander [$?] is 📦 v1.0.0 : 🐍 (warnet) took 27s
₿ echo $status
0

will@ubuntu in ~/src/warnet on  helm-commander [$?] is 📦 v1.0.0 : 🐍 (warnet)
₿ ./test/signet_test.py
2024-09-08 22:14:03 | INFO    | test     | Logging started
2024-09-08 22:14:03 | INFO    | test     | Warnet test base initialized
2024-09-08 22:14:03 | INFO    | test     | Setting up network
2024-09-08 22:14:03 | DEBUG   | test     | Executing warnet command: deploy /home/will/src/warnet/test/data/signet
2024-09-08 22:14:04 | INFO    | test     | Deploying node: miner
Release "miner" does not exist. Installing it now.
NAME: miner
LAST DEPLOYED: Sun Sep  8 22:14:04 2024
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
Thank you for installing miner.
Deploying node: tank-0001
Release "tank-0001" does not exist. Installing it now.
NAME: tank-0001
LAST DEPLOYED: Sun Sep  8 22:14:04 2024
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
Thank you for installing tank-0001.
Deploying node: tank-0002
Release "tank-0002" does not exist. Installing it now.
NAME: tank-0002
LAST DEPLOYED: Sun Sep  8 22:14:04 2024
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
Thank you for installing tank-0002.
2024-09-08 22:14:04 | DEBUG   | test     | Waiting for predicate with timeout 1200s and interval 5s
2024-09-08 22:14:04 | INFO    | test     | Waiting for all tanks to reach 'running': {'total': 3, 'pending': 3}
2024-09-08 22:14:09 | INFO    | test     | Waiting for all tanks to reach 'running': {'total': 3, 'running': 3}
2024-09-08 22:14:09 | DEBUG   | test     | Waiting for predicate with timeout 1200s and interval 5s
Tank miner peers expected: 0, actual: 0
Tank tank-0001 peers expected: 1, actual: 1
Tank tank-0002 peers expected: 1, actual: 1
Network connected
2024-09-08 22:14:10 | DEBUG   | test     | Executing warnet command: bitcoin rpc miner createwallet miner
2024-09-08 22:14:11 | DEBUG   | test     | Executing warnet command: bitcoin rpc miner importdescriptors '[{"desc": "wpkh(tprv8ZgxMBicQKsPfH87iaMtrpzTkWiyFDW7SVWqfsKAhtyEBEqMV6ctPdtc5pNrb2FpSmPcDe8NrxEouUnWj1ud7LT1X1hB1XHKAgB2Z5Z4u2s/84h/1h/0h/0/*)#5j6mshps", "timestamp": 0, "active": true, "internal": false, "range": [0, 999], "next": 0, "next_index": 0}, {"desc": "wpkh(tprv8ZgxMBicQKsPfH87iaMtrpzTkWiyFDW7SVWqfsKAhtyEBEqMV6ctPdtc5pNrb2FpSmPcDe8NrxEouUnWj1ud7LT1X1hB1XHKAgB2Z5Z4u2s/84h/1h/0h/1/*)#9xl6dz3g", "timestamp": 0, "active": true, "internal": true, "range": [0, 999], "next": 0, "next_index": 0}]'
2024-09-08 22:14:11 | DEBUG   | test     | Executing warnet command: run resources/scenarios/signet_miner.py --tank=0 generate --min-nbits --address=tb1q6vakuyw2jhzwmnxcaxryxs6c670fr9esfvrrhj
2024-09-08 22:14:12 | DEBUG   | test     | Waiting for predicate with timeout 300s and interval 5s
2024-09-08 22:14:12 | DEBUG   | test     | Executing warnet command: bitcoin rpc tank-0001 getblockcount
2024-09-08 22:14:17 | DEBUG   | test     | Executing warnet command: bitcoin rpc tank-0001 getblockcount
2024-09-08 22:14:23 | DEBUG   | test     | Executing warnet command: bitcoin rpc tank-0001 getblockcount
2024-09-08 22:14:28 | DEBUG   | test     | Executing warnet command: bitcoin rpc tank-0001 getblockcount
2024-09-08 22:14:34 | DEBUG   | test     | Executing warnet command: bitcoin rpc tank-0001 getblockcount
2024-09-08 22:14:39 | DEBUG   | test     | Executing warnet command: bitcoin rpc tank-0001 getblockcount
2024-09-08 22:14:44 | DEBUG   | test     | Executing warnet command: bitcoin rpc tank-0001 getblockcount
2024-09-08 22:14:50 | DEBUG   | test     | Executing warnet command: bitcoin rpc tank-0001 getblockcount
2024-09-08 22:14:55 | DEBUG   | test     | Executing warnet command: bitcoin rpc tank-0001 getblockcount
2024-09-08 22:15:01 | DEBUG   | test     | Executing warnet command: bitcoin rpc tank-0001 getblockcount
2024-09-08 22:15:06 | DEBUG   | test     | Executing warnet command: bitcoin rpc tank-0001 getblockcount
2024-09-08 22:15:07 | DEBUG   | test     | Executing warnet command: bitcoin rpc tank-0002 getblockcount
2024-09-08 22:15:07 | INFO    | test     | Stopping network
2024-09-08 22:15:07 | DEBUG   | test     | Executing warnet command: down
2024-09-08 22:15:07 | DEBUG   | test     | Waiting for predicate with timeout 60s and interval 1s
2024-09-08 22:15:08 | INFO    | test     | Stopping network
2024-09-08 22:15:08 | DEBUG   | test     | Executing warnet command: down
2024-09-08 22:15:08 | DEBUG   | test     | Waiting for predicate with timeout 60s and interval 1s

will@ubuntu in ~/src/warnet on  helm-commander [$?] is 📦 v1.0.0 : 🐍 (warnet) took 1m5s
₿ ./test/scenarios_test.py
2024-09-08 22:17:27 | INFO    | test     | Logging started
2024-09-08 22:17:27 | INFO    | test     | Warnet test base initialized
2024-09-08 22:17:27 | INFO    | test     | Setting up network
2024-09-08 22:17:27 | DEBUG   | test     | Executing warnet command: deploy /home/will/src/warnet/test/data/12_node_ring
2024-09-08 22:17:31 | INFO    | test     | Deploying node: tank-0000
Release "tank-0000" does not exist. Installing it now.
NAME: tank-0000
LAST DEPLOYED: Sun Sep  8 22:17:27 2024
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
Thank you for installing tank-0000.
Deploying node: tank-0001
Release "tank-0001" does not exist. Installing it now.
NAME: tank-0001
LAST DEPLOYED: Sun Sep  8 22:17:28 2024
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
Thank you for installing tank-0001.
Deploying node: tank-0002
Release "tank-0002" does not exist. Installing it now.
NAME: tank-0002
LAST DEPLOYED: Sun Sep  8 22:17:28 2024
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
Thank you for installing tank-0002.
Deploying node: tank-0003
Release "tank-0003" does not exist. Installing it now.
NAME: tank-0003
LAST DEPLOYED: Sun Sep  8 22:17:29 2024
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
Thank you for installing tank-0003.
Deploying node: tank-0004
Release "tank-0004" does not exist. Installing it now.
NAME: tank-0004
LAST DEPLOYED: Sun Sep  8 22:17:29 2024
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
Thank you for installing tank-0004.
Deploying node: tank-0005
Release "tank-0005" does not exist. Installing it now.
NAME: tank-0005
LAST DEPLOYED: Sun Sep  8 22:17:29 2024
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
Thank you for installing tank-0005.
Deploying node: tank-0006
Release "tank-0006" does not exist. Installing it now.
NAME: tank-0006
LAST DEPLOYED: Sun Sep  8 22:17:29 2024
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
Thank you for installing tank-0006.
Deploying node: tank-0007
Release "tank-0007" does not exist. Installing it now.
NAME: tank-0007
LAST DEPLOYED: Sun Sep  8 22:17:30 2024
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
Thank you for installing tank-0007.
Deploying node: tank-0008
Release "tank-0008" does not exist. Installing it now.
NAME: tank-0008
LAST DEPLOYED: Sun Sep  8 22:17:30 2024
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
Thank you for installing tank-0008.
Deploying node: tank-0009
Release "tank-0009" does not exist. Installing it now.
NAME: tank-0009
LAST DEPLOYED: Sun Sep  8 22:17:30 2024
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
Thank you for installing tank-0009.
Deploying node: tank-0010
Release "tank-0010" does not exist. Installing it now.
NAME: tank-0010
LAST DEPLOYED: Sun Sep  8 22:17:30 2024
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
Thank you for installing tank-0010.
Deploying node: tank-0011
Release "tank-0011" does not exist. Installing it now.
NAME: tank-0011
LAST DEPLOYED: Sun Sep  8 22:17:31 2024
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
Thank you for installing tank-0011.
2024-09-08 22:17:31 | DEBUG   | test     | Waiting for predicate with timeout 1200s and interval 5s
2024-09-08 22:17:31 | INFO    | test     | Waiting for all tanks to reach 'running': {'total': 12, 'running': 6, 'pending': 6}
2024-09-08 22:17:36 | INFO    | test     | Waiting for all tanks to reach 'running': {'total': 12, 'running': 12}
2024-09-08 22:17:36 | DEBUG   | test     | Waiting for predicate with timeout 1200s and interval 5s
Tank tank-0000 peers expected: 1, actual: 1
Tank tank-0001 peers expected: 1, actual: 1
Tank tank-0002 peers expected: 1, actual: 1
Tank tank-0003 peers expected: 1, actual: 1
Tank tank-0004 peers expected: 1, actual: 1
Tank tank-0005 peers expected: 1, actual: 1
Tank tank-0006 peers expected: 1, actual: 1
Tank tank-0007 peers expected: 1, actual: 1
Tank tank-0008 peers expected: 1, actual: 1
Tank tank-0009 peers expected: 1, actual: 1
Tank tank-0010 peers expected: 1, actual: 1
Tank tank-0011 peers expected: 1, actual: 1
Network connected
2024-09-08 22:17:37 | INFO    | test     | Running scenario from file: resources/scenarios/miner_std.py
2024-09-08 22:17:37 | DEBUG   | test     | Executing warnet command: run resources/scenarios/miner_std.py --allnodes --interval=1
2024-09-08 22:17:38 | DEBUG   | test     | Executing warnet command: bitcoin rpc tank-0000 getblockcount
2024-09-08 22:17:38 | DEBUG   | test     | Waiting for predicate with timeout 300s and interval 5s
2024-09-08 22:17:38 | DEBUG   | test     | Waiting for predicate with timeout 300s and interval 5s
2024-09-08 22:17:38 | DEBUG   | test     | Executing warnet command: bitcoin rpc tank-0000 getblockcount
2024-09-08 22:17:39 | DEBUG   | test     | Current block count: 0, target: 2
2024-09-08 22:17:44 | DEBUG   | test     | Executing warnet command: bitcoin rpc tank-0000 getblockcount
2024-09-08 22:17:44 | DEBUG   | test     | Current block count: 1, target: 2
2024-09-08 22:17:49 | DEBUG   | test     | Executing warnet command: bitcoin rpc tank-0000 getblockcount
2024-09-08 22:17:50 | DEBUG   | test     | Current block count: 6, target: 2
2024-09-08 22:17:50 | INFO    | test     | Stopping running scenario
Warning: Immediate deletion does not wait for confirmation that the running resource has been terminated. The resource may continue to run on the cluster indefinitely.
pod "commander-minerstd-1725830258" force deleted
Successfully stopped scenario: commander-minerstd-1725830258
Initiated helm uninstall for release: commander-minerstd-1725830258
2024-09-08 22:17:50 | DEBUG   | test     | Waiting for predicate with timeout 300s and interval 5s
2024-09-08 22:17:50 | DEBUG   | test     | Checking if scenario stopped. Running scenarios: 0
2024-09-08 22:17:50 | INFO    | test     | Running scenario from: test/data/scenario_p2p_interface.py
2024-09-08 22:17:50 | DEBUG   | test     | Executing warnet command: run test/data/scenario_p2p_interface.py
2024-09-08 22:17:51 | DEBUG   | test     | Waiting for predicate with timeout 300s and interval 5s
2024-09-08 22:17:56 | INFO    | test     | Stopping network
2024-09-08 22:17:56 | DEBUG   | test     | Executing warnet command: down
2024-09-08 22:17:56 | DEBUG   | test     | Waiting for predicate with timeout 60s and interval 1s
2024-09-08 22:17:56 | INFO    | test     | Stopping network
2024-09-08 22:17:56 | DEBUG   | test     | Executing warnet command: down
2024-09-08 22:17:57 | DEBUG   | test     | Waiting for predicate with timeout 60s and interval 1s
```

</details>

Would someone mind sanity-checking the tests locally for me on their machine to see if I should bother investigating CI pls?

**Comment by pinheadmz** [comment]
concept ack! Thanks for paying off my tech debt

**Comment by willcl-ark** [comment]
I wonder if it's related to the `build_image` Ci job, which builds a commander image and feeds it to subsequent jobs 🤔

**Comment by willcl-ark** [comment]
The failure here was fixed (in #552) by using my log collector from #553 to find the reason the Commanders were stalling when run in CI:

image_2024-09-09_00-18-31

As seen in the stern logs for the scenario job here: https://github.com/bitcoin-dev-project/warnet/pull/550

**Comment by josibake** [comment]
Concept ACK, thanks for picking this up! will review later today and hopefully we can get this merged

**Comment by pinheadmz** [comment]
LGTM but is this something to worry about?

```
Warning: Immediate deletion does not wait for confirmation that the running resource has been terminated.
The resource may continue to run on the cluster indefinitely.
pod "commander-minerstd-1725890194" force deleted
Successfully stopped scenario: commander-minerstd-1725890194
Initiated helm uninstall for release: commander-minerstd-1725890194
```

File: src/warnet/control.py [hunk 1]
```diff
@@ -77,16 +76,29 @@ def stop(scenario_name):
...
         console.print(f"[bold green]Successfully stopped scenario: {scenario_name}[/bold green]")
     else:
         console.print(f"[bold red]Failed to stop scenario: {scenario_name}[/bold red]")
 
+    # Then uninstall via helm (non-blocking)
+    namespace = get_default_namespace()
+    command = f"helm uninstall {scenario_name} --namespace {namespace} --wait=false"
+
+    # Run the helm uninstall command in the background
+    import subprocess
```
Comment (pinheadmz) [inline src/warnet/control.py]: already imported

File: test/scenarios_test.py [hunk 2]
```diff
@@ -80,7 +80,8 @@ def stop_scenario(self):
         running = scenarios_active()
         assert len(running) == 1, f"Expected one running scenario, got {len(running)}"
         assert running[0]["status"] == "running", "Scenario should be running"
-        delete_pod(running[0]["name"])
+        stop_scenario(running[0]["name"])
+        # delete_pod(running[0]["name"])
```
Comment (pinheadmz) [inline test/scenarios_test.py]: need to keep this?

File: src/warnet/control.py [hunk 3]
```diff
@@ -105,43 +117,53 @@ def list_active_scenarios():
...
+        cmd = f"helm uninstall {release_name} --namespace {namespace} --wait=false"
+        subprocess.Popen(cmd, shell=True, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
+        return f"Initiated uninstall for: {release_name} in namespace {namespace}"
+
+    def delete_pod(pod_name, namespace):
+        cmd = f"kubectl delete pod --ignore-not-found=true {pod_name} -n {namespace} --grace-period=0 --force"
+        subprocess.Popen(cmd, shell=True, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
+        return f"Initiated deletion of pod: {pod_name} in namespace {namespace}"
+
+    with ThreadPoolExecutor(max_workers=10) as executor:
```
Comment (pinheadmz) [inline src/warnet/control.py:142]: Can we use this for `deploy()` as well? a 100 tank network takes a lot longer to start now than it did before helm. I think currently its a foreach/stream_command() but before helm all tanks were started in parallel

File: src/warnet/control.py [hunk 4]
```diff
@@ -184,72 +206,45 @@ def run(scenario_file: str, additional_args: tuple[str]):
...
+            f"scenario={scenario_data}",
+            "--set",
+            f"warnet={warnet_data}",
+        ]
+
+        # Add additional arguments
+        if additional_args:
+            helm_command.extend(["--set", f"args={' '.join(additional_args)}"])
+
+        helm_command.extend([name, "resources/charts/commander"])
```
Comment (pinheadmz) [inline src/warnet/control.py]: this fails when i try to run a scenario from a project directory (not in warnet code repo)

```
(.venv) --> warnet run scenarios/miner_std.py --mature --allnodes --interval 1
Failed to start scenario: miner_std
Error: Error: repo resources not found
```

**Review by pinheadmz — COMMENTED** [review]
I dont think this is fully working yet. Also indicates the need for a test that creates a project directory outside the code repository

**Comment by pinheadmz** [comment]
`warnet down` is really goddamned fast on this branch though! woo

File: src/warnet/control.py (hunk 4 above)
Comment (willcl-ark) [reply src/warnet/control.py]: Thanks nice spot. Need to use `files()`

File: src/warnet/control.py (hunk 1 above)
Comment (willcl-ark) [reply src/warnet/control.py]: Removed

File: src/warnet/control.py (hunk 3 above)
Comment (willcl-ark) [reply src/warnet/control.py:142]: Sure, I don't see why not.

File: test/scenarios_test.py (hunk 2 above)
Comment (willcl-ark) [reply test/scenarios_test.py]: Nope sorry, just a change artefact.

**Review by willcl-ark — COMMENTED** [review]
addressed feedback, thanks

**Pull request merged**